AbigailDawson/knownative
# KN-3: Reusable Component Button

### Describe your changes:

This PR adds a reusable Button component. It contains four variations (PRIMARY, SECONDARY, TERTIARY, AND DANGER). Button props are `buttonText` to label the button, `buttonOnClickFunc` to give the button its onClick action, and `buttonVariant` to identify the style of button. Further iterations could add different sizes as `buttonSize` if developers find a need for it. This PR also refactors the Modal component to use Button as well as replaced the Demo Button on the landing page

![Screenshot 2024-10-29 at 4 34 41 PM](https://github.com/user-attachments/assets/61345a8d-5efa-4ad6-b5e1-8ffdd2e79f12)

### Issue ticket number and link:

https://huly.app/workbench/knownative/tracker/KN-3

### Notes:

None

### How to test:

Check modals and verify button looks and behaves as expected
Check Demo Button on landing page and verify it behaves as expected

<sub><a href="https://huly.app/guest/knownative?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzIyYTI5NDI3Y2M3NjgxYzYyM2MwMTAiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWJpZ2FpbGRhd3NvLWtub3duYXRpdmUtNjYzMjgxYjEtODdiZTE2ZDY5OS0yYTZiY2QifQ.9NzogP9XfYGR2fVYW7JoI9tRa1sNTEtt_ohbNK-0a5o">Huly&reg;: <b>GITHB-152</b></a></sub>
